src/main/java/org/spongepowered/api/plugin/PluginManager.java

```diff
@@ -76,4 +90,53 @@
      */
     boolean isLoaded(String id);
 
+    /**
+     * Gets the id of the {@link Plugin}.
+     *
+     * @return The id
+     */
+    default String getId(Object plugin) {
```

Comment:
There always has to be a limit for these kind of things, and if it results in a lot of duplicated code while not applicable in too many use cases this is exactly the limit IMO. In most cases plugins will want to check several of these methods anyway, and in that case it is much more efficient to call the `find` method once and then call the methods directly.